<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /Users/yury/workspaces/git/youbot-ros-pkg/youbot_common/youbot_description/robots/youbot_gripper.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="youbot" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- The following included files set up definitions of parts of the robot body -->
  <!-- misc common stuff? -->
  <!-- youbot gripper -->
  <!-- generic simulator_gazebo plugins for starting mechanism control, ros time, ros battery -->
  <gazebo>
    <controller:gazebo_ros_controller_manager name="gazebo_ros_controller_manager" plugin="libgazebo_ros_controller_manager.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1000.0</updateRate>
      <interface:audio name="gazebo_ros_controller_manager_dummy_iface"/>
    </controller:gazebo_ros_controller_manager>
    <controller:gazebo_ros_power_monitor name="gazebo_ros_power_monitor_controller" plugin="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <interface:audio name="power_monitor_dummy_interface"/>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </controller:gazebo_ros_power_monitor>
  </gazebo>
  <!-- materials for visualization -->
  <material name="Orange">
    <color rgba="1.0 0.4 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="DarkGrey">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Red">
    <color rgba="1.0 0.0 0.0 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 1.0 0.0 1.0"/>
  </material>
  <!-- Now we can start using the macros included above to define the actual youbot -->
  <!-- The first use of a macro.  This one was defined in base.urdf.xacro above.
       A macro like this will expand to a set of link and joint definitions, and to additional
       Gazebo-related extensions (sensor plugins, etc).  The macro takes an argument, name, 
       that equals "base", and uses it to generate names for its component links and joints 
       (e.g., base_link).  The included origin block is also an argument to the macro.  By convention, 
       the origin block defines where the component is w.r.t its parent (in this case the parent 
       is the world frame). For more, see http://www.ros.org/wiki/xacro -->
  <!-- foot for arm-->
  <link name="base_link">
    
  </link>
  <joint name="gripper_palm_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="gripper_palm_link"/>
  </joint>
  <link name="gripper_palm_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://youbot_description/meshes/youbot_gripper/palm.dae"/>
      </geometry>
      <material name="DarkGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--<box size="0.050 0.094 0.054"/>-->
        <mesh filename="package://youbot_description/meshes/youbot_gripper/palm_convex.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="gripper_finger_joint_l" type="prismatic">
    <origin rpy="0 0 3.14159265359" xyz="0.004 0.008 0.046"/>
    <parent link="gripper_palm_link"/>
    <child link="gripper_finger_link_l"/>
    <axis xyz="0 -1 0"/>
    <dynamics damping="1.0" friction="1.0"/>
    <limit effort="1" lower="0" upper="0.0115" velocity="0.1"/>
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="0.001" soft_upper_limit="0.011"/>
  </joint>
  <link name="gripper_finger_link_l">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.001 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://youbot_description/meshes/youbot_gripper/finger.dae"/>
      </geometry>
      <material name="DarkGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--<box size="0.014 0.020 0.046"/>-->
        <mesh filename="package://youbot_description/meshes/youbot_gripper/finger_convex.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="gripper_finger_joint_r" type="prismatic">
    <origin rpy="0 0 0" xyz="0.004 -0.008 0.046"/>
    <parent link="gripper_palm_link"/>
    <child link="gripper_finger_link_r"/>
    <axis xyz="0 -1 0"/>
    <dynamics damping="1.0" friction="1.0"/>
    <limit effort="1" lower="0" upper="0.0115" velocity="0.1"/>
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="0.001" soft_upper_limit="0.011"/>
  </joint>
  <link name="gripper_finger_link_r">
    <inertial>
      <origin rpy="0 0 0" xyz="0.000 -0.001 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://youbot_description/meshes/youbot_gripper/finger.dae"/>
      </geometry>
      <material name="DarkGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.000 0 0"/>
      <geometry>
        <!--<box size="0.014 0.020 0.046"/>-->
        <mesh filename="package://youbot_description/meshes/youbot_gripper/finger_convex.dae"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="gripper_palm_link">
    <material>Gazebo/Black</material>
    <turnGravityOff>false</turnGravityOff>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="gripper_finger_l_link">
    <material>Gazebo/Black</material>
    <turnGravityOff>false</turnGravityOff>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="gripper_finger_r_link">
    <material>Gazebo/Black</material>
    <turnGravityOff>false</turnGravityOff>
    <selfCollide>true</selfCollide>
  </gazebo>
  <transmission name="gripper_finger_l_trans" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="gripper_finger_l_motor"/>
    <joint name="gripper_finger_joint_l"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <transmission name="gripper_finger_r_trans" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="gripper_finger_r_motor"/>
    <joint name="gripper_finger_joint_r"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
</robot>
